vagrant/provisioning/roles/confluent-platform-install/tasks/main.yml

```diff
@@ -153,6 +153,14 @@
     - "confluent.metrics.topic.replication=1"
   when: not control_center_production_properties_contents.stdout is match('^' + item)
 
+- name: change control center data directory
+  become: yes
+  replace:
+    path: "{{ root_folder }}/app/control-center/control-center-production.properties"
+    regexp: "confluent.controlcenter.data.dir=/var/lib/confluent/control-center"
+    replace: "confluent.controlcenter.data.dir={{ root_folder }}/data/control-center"
+  when: not control_center_production_properties_contents.stdout is match('/var/lib/confluent/control-center')
```

Comment:
The `when` looks inverted to me; you run this task only when /var/lib/confluent/control-center is already not in the file.  Either way it's not needed; the regexp task is idempotent and won't make a change if there's no match.

Reply:
Yes. The when condition is not needed.